src/overlays/actors/ovl_Demo_Getitem/z_demo_getitem.c

```diff
@@ -27,16 +30,88 @@ const ActorInit Demo_Getitem_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+static s16 bankIndex[] = { OBJECT_GI_MASK14, OBJECT_GI_SWORD_4 };
+
+static s16 drawIndex[] = { GID_MASK_GREAT_FAIRY, GID_SWORD_GREAT_FAIRY };
+
+static u16 actionIndex[] = {
+    0x6E,
+    0x236,
+};
+
+void DemoGetitem_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 objIndex;
+    s32 itemIndex;
+    DemoGetitem* this = THIS;
+
+    itemIndex = 0;
+    if ((this->actor.params & 0xF) == 1) {
+        itemIndex = 1;
+    }
+    Actor_SetScale(&this->actor, 0.25f);
+    this->actionFunc = func_80A4FB10;
+    this->item = drawIndex[itemIndex];
+    this->action = actionIndex[itemIndex];
+    objIndex = Object_GetIndex(&globalCtx->objectCtx, bankIndex[itemIndex]);
+    if (objIndex < 0) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        this->object = objIndex;
+    }
+}
+
+void DemoGetitem_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_80A4FB10(DemoGetitem* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->object) != 0) {
+        this->actor.draw = NULL;
+        this->actor.objBankIndex = this->object;
+        this->actionFunc = func_80A4FB68;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Getitem/DemoGetitem_Init.s")
+void func_80A4FB68(DemoGetitem* this, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    s16 sp22 = (globalCtx->gameplayFrames * 0x3E8) & 0xFFFF;
```

Comment:
u16 might be a better type to use, seems to still match (sadly can't get rid of the `0xFFFF`). Might as well use 1000 instead of the hex, since it's more just multiplying the phase rather than an actual angle.